src/glang/codegen/type_resolution.py

```diff
@@ -431,7 +434,9 @@ def pattern_match(
             return None
 
         cost = 0
-        for our_member, target_member in zip(self.members, target.definition.members):
+        for our_member, target_member in zip(
+            self.members, target.definition.members, strict=True
+        ):
```

Comment:
At this point we might as well add our own `zips` helper

Reply:
Idk, sometimes we do actually want `strict=False` (but really the default should be true)